### PR TITLE
MAT-4684 search measures by name or ecqm title

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE PR

Jira Ticket: [MAT-4684](https://jira.cms.gov/browse/MAT-4684)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java [hunk 1]
```diff
@@ -271,4 +271,24 @@ private boolean isMeasurementPeriodChanged(Measure measure, Measure persistedMea
...
+      @RequestParam(required = false, defaultValue = "10", name = "limit") int limit,
+      @RequestParam(required = false, defaultValue = "0", name = "page") int page) {
+
+    final String username = principal.getName();
+    final Pageable pageReq = PageRequest.of(page, limit, Sort.by("lastModifiedAt").descending());
+
+    Page<Measure> measures =
+        filterByCurrentUser
+            ? repository.findAllByMeasureNameOrEcqmTitleForCurrentUser(criteria, pageReq, username)
+            : repository.findAllByMeasureNameOrEcqmTitle(criteria, pageReq);
```
Comment (nmorasb) [inline src/main/java/cms/gov/madie/measure/resources/MeasureController.java:290]: This is fine for now, but as we add more search criteria we may want to move to something like the Criteria class or query by Example.

https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#mongo.query

https://docs.spring.io/spring-data/mongodb/docs/current/reference/html/#query-by-example

**Review by nmorasb — APPROVED** [review]

**Review by jkotanchik-SB — APPROVED** [review]

File: src/main/java/cms/gov/madie/measure/resources/MeasureController.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/cms/gov/madie/measure/resources/MeasureController.java:290]: Ok thanks!
